components/movementsensor/gpsrtkserial/gpsrtkserial_test.go

```diff
@@ -121,18 +128,17 @@ func TestPosition(t *testing.T) {
 	// If there is last error and last position, return last position
 	t.Run("position with last error and last position", func(t *testing.T) {
 		g := &rtkSerial{
-			err:          movementsensor.NewLastError(1, 1),
-			lastposition: movementsensor.NewLastPosition(),
+			err:        movementsensor.NewLastError(1, 1),
+			cachedData: gpsutils.NewCachedData(&mockDataReader{}, logging.NewTestLogger(t)),
 		}
-		g.lastposition.SetLastPosition(testLastPosition)
-
+		initializePosition(g.cachedData)
 		g.err.Set(errors.New("last position"))
-		expectedPos := geo.NewPoint(42.1, 123.)
 
 		pos, alt, err := g.Position(context.Background(), nil)
-		test.That(t, movementsensor.ArePointsEqual(pos, expectedPos), test.ShouldBeTrue)
-		test.That(t, alt, test.ShouldEqual, 0.0)
-		test.That(t, err, test.ShouldBeNil)
+		test.That(t, math.IsNaN(pos.Lat()), test.ShouldBeTrue)
+		test.That(t, math.IsNaN(pos.Lng()), test.ShouldBeTrue)
```

Comment:
This change in the tests makes it look like production behavior has changed, but it's really the test behavior that changed: in production, if there was a `lastErr` to return, we'd return `lastPosition` with it, which was always full of `NaN`s. Only the test could ever set `lastPosition` to anything besides `NaN`. The new version of the test more closely mirrors what has happened all along in production.